### ISSUE TITLE
Ability to specify event icon when creating event with HTTP API

### ISSUE THREAD

**Issue by sebastian-bartkowiak** [blakeblackshear/frigate#12505]
**Describe what you are trying to accomplish and why in non technical terms**
I want to be able to specify event icon when creating event programatically (with HTTP API) so that I can distinguish events easily when reviewing them.

**Describe the solution you'd like**
I belive simple extension of body POSTed at `/api/events/<camera_name>/<label>/create` with field like `icon` would do the trick. That would leave two other things - listing available icons somewhere in the docs for the user to choose from and handling case when user wishes to use unrecognised icon. For the latter I belive a simple fallback to default icon would be acceptable.

**Describe alternatives you've considered**
I don't really see any alternative.

**Additional context**
I'm programatically creating events for multiple things occuring around the house so that I can later easily check them. Since we have event icons in new GUI version it would be cool if I could leverage that :)

**Comment by sebastian-bartkowiak** [comment]
By the way - I noticed minor inconsistency. Event icon is not presented on events listed on top bar of "live" screen, when most recent events are presented.

**Comment by NickM-27** [comment]
this is on purpose, the filmstrip at the top is too small to show the icons because it would cover a large portion of the image

**Comment by sebastian-bartkowiak** [comment]
OK, I think that small icon wouldn't cover that much of the image, but that's not crutial for me.
What do you think about the core of my enhancement idea?

**Comment by NickM-27** [comment]
> OK, I think that small icon wouldn't cover that much of the image, but that's not crutial for me. 

It really does, see this example of a filmstrip item with and without the icons. It covers the white pickup that parked on the street and created the event

![Screen Shot 2024-07-18 at 14 35 15 PM](https://github.com/user-attachments/assets/0c6eeff2-278b-4503-85a5-4ae0af388e56)

![Screen Shot 2024-07-18 at 14 37 23 PM](https://github.com/user-attachments/assets/f5000aad-a587-4c60-803a-f4b4edf384d0)

> What do you think about the core of my enhancement idea?

It would be pretty complicated the way it is described, we do not interface with the events in the UI, it is the review items. So the icon sent to the api, store it as an event, and then store that in reviews but make it clear which item it applies to while we would not want to affect the non custom icons. 

An alternative would be allowing a configuration to create a custom map of custom labels to Fa icons (kind of like how the camera groups works).

**Comment by sebastian-bartkowiak** [comment]
> An alternative would be allowing a configuration to create a custom map of custom labels to Fa icons (kind of like how the camera groups works).

That sounds fine by me - I belive that would additionally allow to change default icon for an event - you could just specify icon for label "person" to be whatever you want it to be :)

**Comment by NickM-27** [comment]
> you could just specify icon for label "person" to be whatever you want it to be :)

right, but that is something we probably don't want, which is where in general stuff like this gets complicated. In general we try to avoid lots of small little options to change small things in the UI because it is more work to maintain, creates more opportunities for bugs, more things to document and keep up to date, etc.

**Comment by sebastian-bartkowiak** [comment]
Yea, I get it, but with #12506 closed this enhancement idea is the only way I see to keep API created events usable. Otherwise all of them have the same icon, and checking their name is not easy (on computer) or not doable at all (on mobile).

**Comment by NickM-27** [comment]
right, I think it will be easier as custom events are used more in the new UI and we can understand what people are using them for and what the best way for the UI to communicate that is
